### DISCUSSION TITLE
Sepia in Italian (female)

### DISCUSSION THREAD

**Discussion by giostark** [SEPIA-Framework/sepia-docs#187] [Show and tell]
**Italians files REV.1**
Ciao Florian!
Finally I managed to produce something of almost useful :-)
I chose the female voice because was the only option and to be honest the robotic voice is nice but not so nice to be useful. 
Probably will be there some small errors: maybe typing errors and also of translation (the meaning sometime is specific of the language and there is not an equivalent) 
I also have to found the equivalent for some services that actually are coded for German locations.
But I swear I'll check on them again and I'll integrate the best I'll can.
I already tried those attached files and they work pretty fair. (with the last revision of the server and the android client).
Tell me what I can translate more. (skin ? other files?)

ps:
i sent you an email time ago , did you received it?

THANKS for your amazing work !

[answers_it.txt](https://github.com/SEPIA-Framework/sepia-docs/files/8734642/answers_it.txt)
[chats_it.txt](https://github.com/SEPIA-Framework/sepia-docs/files/8734643/chats_it.txt)
[teachIt_it.txt](https://github.com/SEPIA-Framework/sepia-docs/files/8734645/teachIt_it.txt)

**Comment by fquirin** [comment]
H Gio,

thanks a lot for the files! :smiley: 
I'll check them out as soon as I can and experiment a bit with the "it" language setting to see what happens :grin: (not that I will understand much ^^ but I just want to look for code errors etc.)

> Tell me what I can translate more. (skin ? other files?)

There are files for the client ([SepiaFW.local](https://github.com/SEPIA-Framework/sepia-html-client-app/blob/master/www/scripts/sepiaFW.local.js)) that need translation sooner or later, but I should probably optimize the code first or the app will always load all texts for all languages at start (not good ^^). Never the less you could have a look already and maybe get familiar with the texts etc..
Ultimately the services and parameters will be the most interesting part ... and the hardest as well since a lot of stuff is hard-coded inside the Java files :see_no_evil: . Maybe for the next update I can pick one service like smart-home and try to rewrite the code so we can load external language files :thinking: , there was some interest anyway to make the 'rooms' and 'smart_devices' parameters extendable .

> ps:
i sent you an email time ago , did you received it?

Yes and I'm very sorry I still haven't answered you :see_no_evil: :-( . I wanted to write a proper reply, but then things became stressful because of the latest update and release and I kept postponing it. I haven't forgotten you though! :slightly_smiling_face:

**Reply by giostark** [reply]
Yo!
Ok fine , many thanks ! :+1: 



An other stuff concerning the language.
If I select Italian of course everything that come read by Sepia is in Italian , and unfortunately also the English words.  
Do you think that is possible to use some variable in the txt file to force Sepia to read in the other language just the words marked?

Ex01
Me: Sepia , come posso migliorarti?
Sepia: Ci sono molti modi etc etc .... prova a guardare sulla pagina di Github.

Now Github come pronounced "Githab" but Sepia read as in italian "Github".
Could be interesting add some option , ex: 
 Sepia: Ci sono molti modi etc etc .... prova a guardare sulla pagina di {en-Github}.
I dont know , as call it as a tag or something similar. 
In this way Sepia could talk right any language.
THEN there is the problem when we talk in other language. Here in Italy we use lots of english words and of course Sepia didnt recognize correctly because she expect italian.
There is an option for multi language in the listening?

**Reply by fquirin** [reply]
> Do you think that is possible to use some variable in the txt file to force Sepia to read in the other language just the words marked?

In theory its possible. Some TTS systems support [SSML](https://www.w3.org/TR/speech-synthesis/) like [Larynx](https://github.com/rhasspy/larynx#ssml) for example (which can be used as drop-in replacement for Mary-TTS.) and it is even in the specs for the Web-Speech-API although in reality support was rather poor. I never really looked into it because it is pretty complicated to handle in a modular system, but some of the SSML commands might actually work in certain cases :thinking: . For anglicisms (English words in native language) SSML could give a specific phoneme spelling, although it would be better if the TTS system was properly trained and had those words in its own vocabulary.

Besides that there is a flag that can be set by SEPIA services to indicate a "one-time language switch" for the next sentence. It's pretty experimental still and it currently only makes sense for news services for example when SEPIA should read the headline of a foreign newspaper etc..
